http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java

```diff
@@ -44,7 +43,7 @@ class DefaultUriBuilder implements UriBuilder {
     private static final String STRING_PATTERN_SCHEME = "([^:/?#]+):";
     private static final String STRING_PATTERN_USER_INFO = "([^@\\[/?#]*)";
     private static final String STRING_PATTERN_HOST_IPV4 = "[^\\[{/?#:]*";
-    private static final String STRING_PATTERN_HOST_IPV6 = "\\[[\\p{XDigit}\\:\\.]*[%\\p{Alnum}]*\\]";
+    private static final String STRING_PATTERN_HOST_IPV6 = "\\[[\\p{XDigit}:.]*[%\\p{Alnum}]*]";
```

Comment:
can you explain the changes to these patterns?

Reply:
Just redundant characters escape. This is just a simplification of the expression, removed unnecessary escapes